File: FlexTargetiOS/flextarget/View/Drills/GameDrillResultView.swift
import SwiftUI

struct GameDrillResultView: View {
    let gameName: String
    let score: String
    let hits: String
    let misses: String
    var onReplay: () -> Void
    var onDone: () -> Void
    
    private let accentRed = Color(red: 0.8705882352941177, green: 0.2196078431372549, blue: 0.13725490196078433)
    private let darkBackground = Color.black
    private let darkText = Color(red: 0.098, green: 0.098, blue: 0.098) // #191919

    var body: some View {
        ZStack {
            darkBackground.ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Custom Navigation Bar
                HStack {
                    Button(action: onDone) {
                        Image(systemName: "chevron.left")
                            .font(.system(size: 20, weight: .semibold))
                            .foregroundColor(accentRed)
                    }
                    Spacer()
                    Text(NSLocalizedString("game_results", comment: "Game Results"))
                        .font(.system(size: 20, weight: .semibold))
                        .foregroundColor(accentRed)
                    Spacer()
                    // Hidden placeholder for symmetry
                    Image(systemName: "chevron.left")
                        .font(.system(size: 20, weight: .semibold))
                        .foregroundColor(.clear)
                }
                .padding(.horizontal, 20)
                .padding(.top, 10)
                .padding(.bottom, 20)
                
                VStack(spacing: 0) {
                    Spacer()
                    
                    // Header Info
                    VStack(spacing: 8) {
                        Text(gameName.uppercased())
                            .font(.system(size: 18, weight: .bold))
                            .foregroundColor(accentRed)
                        
                        Spacer().frame(height: 40)
                        
                        // Big Score Display
                        Text(NSLocalizedString("score", comment: "Score").uppercased())
                            .font(.system(size: 14, weight: .medium))
                            .foregroundColor(.gray)
                        
                        Text(score)
                            .font(.system(size: 72, weight: .black))
                            .foregroundColor(accentRed)
                            .multilineTextAlignment(.center)
                    }
                    
                    Spacer()
                    
                    // Stats Row
                    HStack {
                        Spacer()
                        GameStatItem(label: NSLocalizedString("hits", comment: "Hits").uppercased(), value: hits, color: .green)
                        Spacer()
                        GameStatItem(label: NSLocalizedString("misses", comment: "Misses").uppercased(), value: misses, color: accentRed)
                        Spacer()
                    }
                    .padding(.horizontal, 24)
                    
                    Spacer()
                    
                    // Buttons
                    VStack(spacing: 16) {
                        Button(action: onReplay) {
                            HStack(spacing: 8) {
                                Image(systemName: "arrow.clockwise")
                                    .font(.system(size: 18, weight: .bold))
                                Text(NSLocalizedString("replay", comment: "Replay").uppercased())
                                    .font(.system(size: 18, weight: .bold))
                            }
                            .foregroundColor(darkText)
                            .frame(maxWidth: .infinity)
                            .frame(height: 56)
                            .background(accentRed)
                            .cornerRadius(12)
                        }
                        
                        Button(action: onDone) {
                            Text(NSLocalizedString("done", comment: "Done").uppercased())
                                .font(.system(size: 18, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .frame(height: 56)
                                .background(
                                    RoundedRectangle(cornerRadius: 12)
                                        .stroke(Color.gray, lineWidth: 1)
                                )
                        }
                    }
                    .padding(.horizontal, 24)
                    .padding(.bottom, 60) // Move buttons up as requested
                }
            }
        }
        .navigationBarHidden(true)
    }
}

struct GameStatItem: View {
    let label: String
    let value: String
    let color: Color
    
    var body: some View {
        VStack(spacing: 4) {
            Text(label)
                .font(.system(size: 12, weight: .medium))
                .foregroundColor(.gray)
            Text(value)
                .font(.system(size: 32, weight: .black))
                .foregroundColor(color)
        }
    }
}

File: FlexTargetAndroid/app/src/main/java/com/flextarget/android/ui/drills/GameDrillResultView.kt
package com.flextarget.android.ui.drills

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.flextarget.android.ui.theme.ttNormFontFamily

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun GameDrillResultView(
    gameName: String,
    score: String,
    hits: String,
    misses: String,
    onReplay: () -> Unit = {},
    onDone: () -> Unit
) {
    val accentRed = Color(0xffde3823)
    val darkBackground = Color.Black

    Scaffold(
        topBar = {
            CenterAlignedTopAppBar(
                title = {
                    Text(
                        text = "GAME RESULTS",
                        color = accentRed,
                        fontSize = 20.sp,
                        fontWeight = FontWeight.SemiBold,
                        fontFamily = ttNormFontFamily
                    )
                },
                navigationIcon = {
                    IconButton(onClick = onDone) {
                        Icon(
                            Icons.Default.ArrowBack,
                            contentDescription = "Back",
                            tint = accentRed
                        )
                    }
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
                    containerColor = darkBackground
                )
            )
        },
        containerColor = darkBackground
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(24.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.SpaceBetween
        ) {
            // Header Info
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text(
                    text = gameName.uppercase(),
                    color = accentRed,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    fontFamily = ttNormFontFamily
                )
                Spacer(modifier = Modifier.height(40.dp))
                
                // Big Score Display
                Text(
                    text = "SCORE",
                    color = Color.Gray,
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Medium,
                    fontFamily = ttNormFontFamily
                )
                Text(
                    text = score,
                    color = accentRed,
                    fontSize = 72.sp,
                    fontWeight = FontWeight.Black,
                    fontFamily = ttNormFontFamily,
                    textAlign = TextAlign.Center
                )
            }

            // Stats Row
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {
                GameStatItem(label = "HITS", value = hits, color = Color.Green)
                GameStatItem(label = "MISSES", value = misses, color = accentRed)
            }

            // Buttons
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 60.dp),
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                Button(
                    onClick = onReplay,
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(56.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = accentRed),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Icon(
                        Icons.Default.Refresh,
                        contentDescription = null,
                        tint = Color(0xff191919)
                    )
                    Spacer(Modifier.width(8.dp))
                    Text(
                        "REPLAY",
                        color = Color(0xff191919),
                        fontSize = 18.sp,
                        fontWeight = FontWeight.Bold,
                        fontFamily = ttNormFontFamily
                    )
                }

                OutlinedButton(
                    onClick = onDone,
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(56.dp),
                    border = androidx.compose.foundation.BorderStroke(1.dp, Color.Gray),
                    shape = RoundedCornerShape(12.dp),
                    colors = ButtonDefaults.outlinedButtonColors(contentColor = Color.White)
                ) {
                    Text(
                        "DONE",
                        fontSize = 18.sp,
                        fontWeight = FontWeight.Bold,
                        fontFamily = ttNormFontFamily
                    )
                }
            }
        }
    }
}

@Composable
fun GameStatItem(label: String, value: String, color: Color) {
    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Text(
            text = label,
            color = Color.Gray,
            fontSize = 12.sp,
            fontWeight = FontWeight.Medium,
            fontFamily = ttNormFontFamily
        )
        Text(
            text = value,
            color = color,
            fontSize = 32.sp,
            fontWeight = FontWeight.Black,
            fontFamily = ttNormFontFamily
        )
    }
}
